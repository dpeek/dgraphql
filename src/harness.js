import fs from 'fs'
import path from 'path'

import { graphql } from 'graphql'

import { buildSchema } from './schema'

import type { ClientConfig } from './client'

export async function init (config: ClientConfig) {
  const schemaPath = path.resolve(__dirname, '__tests__/schema.graphql')
  const sourcePath = path.resolve(__dirname, '__tests__/source.graphql')

  const source = fs.readFileSync(sourcePath).toString()

  const time = String(new Date().getTime() - 1495660000000)
  let commonVariables = { time }

  // can't use the same schema as some tests are in relay mode, the query is not
  const dataSchema = buildSchema(fs.readFileSync(schemaPath).toString(), {})
  const data = await graphql({
    schema: dataSchema,
    source,
    variableValues: commonVariables,
    contextValue: { language: 'en' }
  })
  const walk = node => {
    if (node.id && node.name) {
      commonVariables[node.name.split(' ')[0].toLowerCase()] = node.id
    }
    Object.values(node).forEach(value => {
      if (typeof value === 'object') walk(value)
    })
  }
  walk(data)

  const schema = buildSchema(fs.readFileSync(schemaPath).toString(), config)
  return (source, variables, language) => {
    return graphql({
      schema,
      source,
      variableValues: { ...commonVariables, ...(variables || {}) },
      contextValue: { language: language || 'en' }
    })
  }
}
