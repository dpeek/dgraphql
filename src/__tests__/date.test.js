import { init } from './harness'

var graphql

beforeAll(async () => {
  graphql = await init()
})

test('sets Date and DateTime as String', async () => {
  const mutation = `mutation {
    createPerson(input: {
      dob: "2000-12-01",
      lastActiveAt: "2000-12-01"
    }) {
      person {
        dob
        lastActiveAt
      }
    }
  }`
  const result = await graphql(mutation)
  expect(result).toMatchSnapshot()
})
