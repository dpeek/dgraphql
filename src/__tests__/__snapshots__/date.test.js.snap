// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets Date and DateTime as String 1`] = `
Object {
  "data": Object {
    "createPerson": Object {
      "person": Object {
        "dob": "2000-12-01T00:00:00Z",
        "lastActiveAt": "2000-12-01T00:00:00Z",
      },
    },
  },
}
`;
